### PR TITLE
NixOS in container

### LINKED ISSUE #8: NixOS-based image support
Using NixOS inside a service is already possible, but not well integrated into Arion. A 'standard' interface seems helpful because it should make switching between init systems easier. For instance, it creates a standard way to add NixOS options to docker-compose services.

 - Service-level option for NixOS configuration to be written by user
    - cast to list of modules
    - merge function is list concatenation
 - Service-level option for evaluated NixOS configuration to be read by user or internal module
    - defined by calling `pkgs.nixos` function with these modules
 - ~Internal option and module for nlewo's [nix-container-images](https://github.com/nlewo/nix-container-images)~ moved to #24
    - may or may not be able to use generalized call to NixOS
    - may need to provide extra module to nix-container-images in order to support `useHostStore` properly
 - Internal option and module for `services.systemd.*.runner`-based containers

### PR DESCRIPTION
**Pull request by roberth**
Closes #8

### CODE REVIEW

File: src/nix/modules/service/nixos-init.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+  Invokes the NixOS init system in the container.
+ */
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options = {
+    nixos.useInit = lib.mkOption {
```
Comment (domenkozar) [inline src/nix/modules/service/nixos-init.nix]: I'd call this `useSystemd`  and have different options if we decide to support a different init in the future.

File: examples/full-nixos/arion-compose.nix [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  docker-compose.services.webserver = { pkgs, ... }: {
+    nixos.useInit = true;
+    nixos.configuration.services.nginx.enable = true;
+    nixos.configuration.services.nginx.virtualHosts.localhost.root = "${pkgs.nix.doc}/share/doc/nix/manual";
+    service.useHostStore = true;
```
Comment (domenkozar) [inline examples/full-nixos/arion-compose.nix]: Why is this needed? Maybe it should be turned on by default?

File: examples/nixos-unit/arion-compose.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  One known issue is that the script does not respond to docker's
+  SIGTERM shutdown signal.
+
+ */
+
+{
+  docker-compose.services.webserver = { config, pkgs, ... }: {
+
+    nixos.configuration = {config, pkgs, ...}: {
+      boot.isContainer = true;
```
Comment (domenkozar) [inline examples/nixos-unit/arion-compose.nix]: Shouldn't this be part of the profile in `container-systemd.nix`

Comment (roberth) [reply examples/nixos-unit/arion-compose.nix]: No, it doesn't use systemd.

File: examples/full-nixos/arion-compose.nix (hunk 2 above)
Comment (roberth) [reply examples/full-nixos/arion-compose.nix]: It's not strictly needed, but it is currently the only way to do Nix-based images :)
The alternative will be to integrate dockerTools, #6
